### PR TITLE
7.0 add web note and web note test

### PR DESCRIPTION
**Pull request by ntodorova**
This module can be used for adding a notes field in any module that need them.
There are three type of notes - private, internal, external. 
Only the user that create a private note can see it. The other two types can be used for creating different views.
There is also a test module that shows use case.

### CODE REVIEW

**Comment by ntodorova** [comment]
@rami-wafaie would you please review this?

**Comment by OSguard** [comment]
2016-01-08 21:40:40,751 10470 WARNING 3139434-216-bd51b9-all openerp.modules.loading: The model web.note has no access rules, consider adding one. E.g. access_web_note,access_web_note,model_web_note,,1,1,1,1
2016-01-08 21:40:40,751 10470 WARNING 3139434-216-bd51b9-all openerp.modules.loading: The model web.note.container has no access rules, consider adding one. E.g. access_web_note_container,access_web_note_container,model_web_note_container,,1,1,1,1

should be fixed

**Comment by rami-wafaie** [comment]
:thumbsup:

File: web_note/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Web Note',
+    'version': '0.1',
```
Comment (OSguard) [inline web_note/__openerp__.py]: 7.0.1.0

**Comment by hbrunn** [comment]
nice idea. Why don't you provide an abstract model that has some predefined notes field, so that users of this function only need to say

``` python
class MyModel(models.Model):
    _inherit = ['original.model', 'your.abstract.model']
```

and adapt their views? That's a lot more convenient imho

File: web_note_test/__init__.py [hunk 2]
```diff
@@ -0,0 +1,23 @@
+# -*- coding: utf-8 -*-
```
Comment (hbrunn) [inline web_note_test/__init__.py]: don't add a demo module on this level. Put it into your actual module, if somebody want to try this out, this person can symlink it into the addons path

**Comment by pedrobaeza** [comment]
There are comments unattended. Is there anyone interested in this?

**Comment by ntodorova** [comment]
Yes, I would like to finish that one. @hbrunn I was just confused about your comment, what is your idea for abstract module with o2m field, how should the connection be implemented?

**Comment by hbrunn** [comment]
don't worry about this for now, let's merge first

**Comment by ntodorova** [comment]
Ok, so then from my side it is ready. What do we need so we can merge it?

File: web_note/README.rst [hunk 3]
```diff
@@ -0,0 +1,36 @@
+Web Note
```
Comment (pedrobaeza) [inline web_note/README.rst:6]: Please use latest README template: https://raw.githubusercontent.com/OCA/maintainer-tools/master/template/module/README.rst

File: web_note/README.rst [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+================
+
+This module can be used for adding a notes field in any module that need them.
+There are three type of notes - private, internal, external. 
+Only the user that create a private note can see it. The other two types can be used for creating different views.
+
+Installation
+============
+
+Installing the module itself is enough.
```
Comment (pedrobaeza) [inline web_note/README.rst]: Remove useless section like this one or configuration

File: web_note/README.rst [hunk 5]
```diff
@@ -0,0 +1,36 @@
...
+To use this module, you need to:
+
+ * Add dependencie to 'web_note' in the __openerp__.py file of the module in which you need the webnotes(your module).
+ * inherit the web.note class and add many2one field connected with your model
+ * in your model create one2many field to web.note model
+ * add the field in the view you want
+ 
+
+
+Versions
```
Comment (pedrobaeza) [inline web_note/README.rst]: This section is not one of the valid ones.

File: web_note/__init__.py [hunk 6]
```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline web_note/__init__.py]: Please short headers like in https://github.com/OCA/maintainer-tools/blob/master/template/module/__init__.py

File: web_note/__openerp__.py [hunk 7]
```diff
@@ -0,0 +1,55 @@
...
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+{
+    'name': 'Web Note',
+    'version': '7.0.1.0',
```
Comment (pedrobaeza) [inline web_note/__openerp__.py]: It should be 7.0.1.0.0

File: web_note/__openerp__.py [hunk 8]
```diff
@@ -0,0 +1,55 @@
...
+    'website': 'http://www.initos.com',
+    'license': 'AGPL-3',
+    'images': [],
+    'depends': [],
+    'data': [
+        "web_note_view.xml",
+        "security/web_note_security.xml",
+        "security/ir.model.access.csv",
+    ],
+    'js': [
```
Comment (pedrobaeza) [inline web_note/__openerp__.py]: Remove empty keys

File: web_note/web_note.py [hunk 9]
```diff
@@ -0,0 +1,116 @@
...
+    p = re.compile(r'\s+')
+    return p.sub(' ', data)
+
+
+def _type_selection(self, *args, **kwargs):
+    """Redirect for easier overwriting."""
+    return self.pool.get('web.note').type_selection(*args, **kwargs)
+
+
+class web_note(orm.Model):
```
Comment (pedrobaeza) [inline web_note/web_note.py]: CamelCase for class name: `WebNote`

File: web_note/web_note.py [hunk 10]
```diff
@@ -0,0 +1,116 @@
...
+        'sequence': fields.integer('Order in report'),
+        'create_uid': fields.many2one('res.users', 'User', readonly=True),
+    }
+
+    _defaults = {
+        'sequence': 10,
+    }
+
+
+class web_note_container(orm.Model):
```
Comment (pedrobaeza) [inline web_note/web_note.py]: CamelCase

File: web_note/web_note.py [hunk 11]
```diff
@@ -0,0 +1,116 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline web_note/web_note.py]: Short header

File: web_note/__openerp__.py [hunk 12]
```diff
@@ -0,0 +1,55 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (pedrobaeza) [inline web_note/__openerp__.py]: Short header

File: web_note/web_note_view.xml [hunk 13]
```diff
@@ -0,0 +1,32 @@
...
+                <tree string="Notes" version="7.0">
+                    <field name="sequence" widget="handle"/>
+                    <field name="type"/>
+                    <field name="container_id"/>
+                    <field name="display_name"/>
+                </tree>
+            </field>
+        </record>
+    </data>
+</openerp>
```
Comment (pedrobaeza) [inline web_note/web_note_view.xml]: Linebreak at the end

File: web_note_test/__init__.py (hunk 2 above)
Comment (pedrobaeza) [reply web_note_test/__init__.py]: I agree this should go in the main module, as res.partner is a base model.

File: web_note_test/web_note.py [hunk 14]
```diff
@@ -0,0 +1,30 @@
...
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+from openerp.osv import orm, fields
+
+
+class web_notes(orm.Model):
+    _inherit = 'web.note'
+    _columns = {
+        'res_partner_id':
```
Comment (pedrobaeza) [inline web_note_test/web_note.py]: Usual name for this field is `partner_id`. You should rename it

**Review by pedrobaeza — CHANGES_REQUESTED** [review]

**Comment by ntodorova** [comment]
@pedrobaeza I have made the changes you required.

**Review by pedrobaeza — APPROVED** [review]
Thanks for the changes

**Pull request merged**